### ISSUE TITLE
[BUG] to_orc with snappy compression fails to compress the files

### ISSUE THREAD

**Issue by mlahir1** [rapidsai/cudf#12170]
to_orc with snappy compression fails to compress the files

### Steps to repro.
1. Loading an orc file 124 MB into cudf df
2. writing back with snappy and without compressions.

Both the file sizes are inflated and about the same (223MB and 224MB)

```
df = cudf.read_orc("test.orc", use_index=False)
df.to_orc("test_snappy.orc", compression='snappy')
df.to_orc("test_nc.orc", compression=None)

$ ls | grep orc | grep test
-rw-r--r--. 1 root root 124M Nov 16 15:30 test.orc
-rw-r--r--. 1 root root 223M Nov 16 18:10 test_nc.orc
-rw-r--r--. 1 root root 224M Nov 16 18:10 test_snappy.
```

### Expected behavior, 
The same above experiment performed with rapids 22.06 gives a compressed file size. However it is still inflated and > 124MB
```
-rw-r--r--. 1 root root 124M Nov 16 15:30 test.orc
-rw-r--r--. 1 root root 223M Nov 16 18:17 test_nc.orc
-rw-r--r--. 1 root root 168M Nov 16 18:17 test_snappy.orc
```

**Comment by GregoryKimball** [comment]
Thank you @mlahir1 for raising this issue. It looks like we have a logic error blocking the compressor from running. We expect that #12194 will resolve this issue.

**Comment by GregoryKimball** [comment]
Closed by #12194
